src/Microsoft.AspNetCore.Proxy/ProxyAdvancedExtensions.cs

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Proxy
+{
+    public static class ProxyAdvancedExtentions
+    {
+        private static readonly string[] NotForwardedWebSocketHeaders = new[] { "Connection", "Host", "Upgrade", "Sec-WebSocket-Key", "Sec-WebSocket-Version" };
+        private const int DefaultWebSocketBufferSize = 4096;
+
+        public static Uri ToWebSocketScheme(this Uri uri)
```

Comment:
Rotfl ;) - total misunderstanding. Funny part is that I have ended up inheriting DelegatingHandler and the code seems much cleaner.

Now I would be the one opting for internalizing entire ProxyAdvancedExtensions... as cannot think of any scenario that couldn't be nicely handled by the DelegatingHandler.